engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -838,6 +874,707 @@ static int SetTexture(lua_State* L)
+    ValidateSetAtlasAnimations(L, &num_animations);
+
+    if (num_animations > 0 && num_geometries == 0)
+    {
+        lua_pop(L, 1);
+        return DM_LUA_ERROR("Setting atlas with animations requires geometry");
+    }
+
+    texture_set_ddf.m_Geometries.m_Data  = new dmGameSystemDDF::SpriteGeometry[num_geometries];
+    texture_set_ddf.m_Geometries.m_Count = num_geometries;
+    memset(texture_set_ddf.m_Geometries.m_Data, 0, sizeof(dmGameSystemDDF::SpriteGeometry) * num_geometries);
+
+    texture_set_ddf.m_Animations.m_Data  = new dmGameSystemDDF::TextureSetAnimation[num_animations];
+    texture_set_ddf.m_Animations.m_Count = num_animations;
+    memset(texture_set_ddf.m_Animations.m_Data, 0, sizeof(dmGameSystemDDF::TextureSetAnimation) * num_animations);
+
+    if (num_geometries > 0)
+    {
+        float inv_tex_width  = 1.0f / tex_width;
+        float inv_tex_height = 1.0f / tex_height;
+
+        lua_getfield(L, -1, "geometries");
+        for (int i = 0; i < num_geometries; ++i)
+        {
+            lua_pushnumber(L, i+1);
+            lua_gettable(L, -2);
+            assert(lua_istable(L, -1));
+
+            dmGameSystemDDF::SpriteGeometry& geometry = texture_set_ddf.m_Geometries[i];
+            MakeNumberArrayFromLuaTable<float>(L, "vertices", (void**) &geometry.m_Vertices.m_Data, &geometry.m_Vertices.m_Count);
+            MakeNumberArrayFromLuaTable<float>(L, "uvs", (void**) &geometry.m_Uvs.m_Data, &geometry.m_Uvs.m_Count);
+            MakeNumberArrayFromLuaTable<int>(L, "indices", (void**) &geometry.m_Indices.m_Data, &geometry.m_Indices.m_Count);
+
+            lua_pop(L, 1);
+
+            // Check that each stream has the correct amount of components
+            #define VALIDATE_STREAM_LENGTH(stream_name, stream_length, num_components) \
+                if (stream_length % num_components != 0) \
+                { \
+                    lua_pop(L, 2); \
+                    DestroyTextureSet(texture_set_ddf); \
+                    return DM_LUA_ERROR("Uneven number of entries in %s table for geometry [%d]", stream_name, i); \
+                }
+            VALIDATE_STREAM_LENGTH("vertices", geometry.m_Vertices.m_Count, 2);
+            VALIDATE_STREAM_LENGTH("uvs", geometry.m_Uvs.m_Count, 2);
+            VALIDATE_STREAM_LENGTH("indices", geometry.m_Indices.m_Count, 3);
+            #undef VALIDATE_STREAM_LENGTH
+
+            // Calculate extents so that we can transform to -0.5 .. 0.5 based
+            // on the middle of the sprite
+            float geo_width = 0.0f;
+            float geo_height = 0.0f;
+            for (int j = 0; j < geometry.m_Vertices.m_Count; j += 2)
+            {
+                geo_width  = dmMath::Max(geo_width, geometry.m_Vertices.m_Data[j]);
+                geo_height = dmMath::Max(geo_height, geometry.m_Vertices.m_Data[j+1]);
+            }
+
+            geometry.m_Width  = geo_width;
+            geometry.m_Height = geo_height;
```

Comment:
The alternative to this would be to pass in width/height explicitly _per_ geometry, but this seems to work so I think it's a nicer ux